maven-artifact/src/main/java/org/apache/maven/artifact/versioning/DefaultArtifactVersion.java

```diff
@@ -204,19 +211,33 @@ public final void parseVersion( String version )
     }
 
     private static Integer getNextIntegerToken( StringTokenizer tok )
+    {
+        String s = tok.nextToken();
+        if ( ( s.length() > 1 ) && s.startsWith( "0" ) )
+        {
+            return null;
+        }
+        return tryParseInt( s );
+    }
+
+    private static Integer tryParseInt( String s )
     {
         try
         {
-            String s = tok.nextToken();
-            if ( ( s.length() > 1 ) && s.startsWith( "0" ) )
+            if ( !isParsable( s ) )
+            {
+                return null;
+            }
+            long longValue = Long.parseLong( s );
+            if ( longValue > Integer.MAX_VALUE )
```

Comment:
Again, change in semantics with long. Dislike on this one. Why all the hassle, why not use  `NumberUtils` from Commons Lang 3?

Reply:
Because it throws an exception internally, which is what we're trying to avoid in the first place here. We don't want to throw exceptions on common code paths.

Reply:
+1 to @oehme, however it is probably worth adding a comment that explains this (the reason behind `tryParseInt` method and the reason behind `parseLong`).